x-pack/platform/plugins/shared/task_manager/server/lib/get_first_run_at.test.ts

```diff
@@ -0,0 +1,300 @@
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockLogger } from '../test_utils';
+import { getFirstRunAt } from './get_first_run_at';
+
+describe('getFirstRunAt', () => {
+  const logger = mockLogger();
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date('2025-04-15T13:01:02Z'));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.clearAllTimers();
+  });
+
+  test('should return runAt when provided', () => {
+    const runAt = new Date(Date.now() + 3000);
+    const taskInstance = {
+      id: 'id',
+      params: {},
+      state: {},
+      taskType: 'report',
+      runAt,
+    };
+
+    const firstRunAt = getFirstRunAt({ taskInstance, logger });
+    expect(firstRunAt).toEqual(runAt.toISOString());
+  });
+
+  test('should return now when there is neither runAt nor rrule provided ', () => {
+    const now = new Date();
+    const taskInstance = {
+      id: 'id',
+      params: {},
+      state: {},
+      taskType: 'report',
+    };
+    const firstRunAt = getFirstRunAt({ taskInstance, logger });
+    expect(firstRunAt).toEqual(now.toISOString());
+  });
+
+  test('should return now when an rrule with simple interval is provided ', () => {
+    const now = new Date();
+    const taskInstance = {
+      id: 'id',
+      params: {},
+      state: {},
+      taskType: 'report',
+      schedule: {
+        rrule: {
+          freq: 3,
+          interval: 1,
+          tzid: 'UTC',
+        },
+      },
+    };
+    const firstRunAt = getFirstRunAt({ taskInstance, logger });
+    expect(firstRunAt).toEqual(now.toISOString());
+  });
+
+  test('should return now when a simple interval is provided ', () => {
+    const now = new Date();
+    const taskInstance = {
+      id: 'id',
+      params: {},
+      state: {},
+      taskType: 'report',
+
+      schedule: {
+        interval: '1m',
+      },
+    };
+    const firstRunAt = getFirstRunAt({ taskInstance, logger });
+    expect(firstRunAt).toEqual(now.toISOString());
+  });
+
+  test('should return the calculated runAt when an rrule with fixed time is provided', () => {
+    const taskInstance = {
+      id: 'id',
+      params: {},
+      state: {},
+      taskType: 'report',
+      schedule: {
+        rrule: {
+          freq: 3,
+          interval: 1,
+          tzid: 'UTC',
+          byhour: [12],
+          byminute: [15],
+        },
+      },
+    };
+    const firstRunAt = getFirstRunAt({ taskInstance, logger });
+    const firstRunAtDate = new Date(firstRunAt);
+    // The next day from 2025-04-15 is 2025-04-16
+    // The time is set to 12:15
+    expect(firstRunAtDate).toEqual(new Date('2025-04-16T12:15:00Z'));
+  });
+
+  test('should return the calculated runAt when an rrule only with byhour is provided', () => {
```

Comment:
nit (also for similarly named tests below)
```suggestion
  test('should return the calculated runAt when an rrule with only byhour is provided', () => {
```